lib/parse_date/int_from_string.rb

```diff
@@ -89,57 +81,43 @@ def sortable_year_for_yy(orig_date_str)
 
     DECADE_4CHAR_REGEXP = Regexp.new('(^|\D)\d{3}[u\-?x]')
 
-    # get first year of decade (as String) if we have:  yyyu, yyy-, yyy? or yyyx pattern
+    # first year of decade (as String) if we have:  yyyu, yyy-, yyy? or yyyx pattern
     #   note that these are the only decade patterns found in our actual date strings in MODS records
     # @return [String, nil] 4 digit year (e.g. 1860, 1950) if orig_date_str matches pattern, nil otherwise
-    def sortable_year_for_decade(orig_date_str)
+    def first_year_for_decade(orig_date_str)
       decade_matches = orig_date_str.match(DECADE_4CHAR_REGEXP) if orig_date_str
       changed_to_zero = decade_matches.to_s.tr('u\-?x', '0') if decade_matches
-      ParseDate.sortable_year_for_yyyy(changed_to_zero) if changed_to_zero
+      ParseDate.first_four_digits(changed_to_zero) if changed_to_zero
     end
 
     CENTURY_WORD_REGEXP = Regexp.new('(\d{1,2}).*century')
     CENTURY_4CHAR_REGEXP = Regexp.new('(\d{1,2})[u\-]{2}([^u\-]|$)')
 
-    # get first year of century (as String) if we have:  yyuu, yy--, yy--? or xxth century pattern
+    # first year of century (as String) if we have:  yyuu, yy--, yy--? or xxth century pattern
     #   note that these are the only century patterns found in our actual date strings in MODS records
     # @return [String, nil] yy00 if orig_date_str matches pattern, nil otherwise; also nil if B.C. in pattern
-    def sortable_year_for_century(orig_date_str)
+    def first_year_for_century(orig_date_str)
       return unless orig_date_str
       return if orig_date_str =~ /B\.C\./
-
-      century_matches = orig_date_str.match(CENTURY_4CHAR_REGEXP)
-      if century_matches
-        m = Regexp.last_match(1)
-        return m + '00' if m.length == 2
-        return '0' + m + '00' if m.length == 1
-      end
-
-      century_str_matches = orig_date_str.match(CENTURY_WORD_REGEXP)
-      return unless century_str_matches
-
-      yy = (Regexp.last_match(1).to_i - 1).to_s
-      return yy + '00' if yy.length == 2
-      return '0' + yy + '00' if yy.length == 1
```

Comment:
since we are now concerned with integers, rather than sortable strings (ew), the whole leading zero complexity goes away.